### ISSUE TITLE
Add more information on map resize

### ISSUE THREAD

**Issue by colleenmcginnis** [mapbox/mapbox-gl-js-docs#65]
Several users have provided feedback on the [`resize` section](https://docs.mapbox.com/mapbox-gl-js/api/#map#resize) of the API reference docs. There is very little information about when to use it and no information on how to use it. All that I could find was @danswick's [gist from 2016](https://gist.github.com/danswick/fc56f37c10d40be62e4feac5984250d2). 

Can someone from @mapbox/gl-js team help me understand how `resize` works (including details on the `eventData` parameter is intended to be used) so I can add an example to the docs?

**Comment by mourner** [comment]
Perhaps it would be useful to reword it to reflect when it should be used. Here's an equivalent method from Leaflet docs:

> Checks if the map container size changed and updates the map if so — call it after you've changed the map size dynamically

Any `eventData` is simply passed on to `movestart`, `move`, `resize` and `moveend` events that get triggered as a result of resize. As far as I remember, this is sometimes useful for differentiating the source of an event (e.g. user-initiated or programmatically triggered).
